management/src/main/java/org/kaazing/gateway/management/monitoring/configuration/impl/MonitorFileWriterImpl.java

```diff
@@ -0,0 +1,321 @@
+    }
+
+    /**
+     * Creates the gateway counter labels buffer
+     * @param buffer - the underlying byte buffer
+     * @param metaDataBuffer - the meta data buffer from which we compute the offset
+     * @return the counter labels buffer
+     */
+    @Override
+    public UnsafeBuffer createGatewayCounterLabelsBuffer(final ByteBuffer buffer) {
+        final int offset = endOfMetadata;
+        final int length = metaDataBuffer.getInt(gwCountersLblBuffersLengthOffset);
+        // Update offset in header section
+        metaDataBuffer.putInt(gwCountersLblBuffersReferenceOffset, offset);
+
+        return new UnsafeBuffer(buffer, offset, length);
+    }
+
+    /**
+     * Creates the gateway counter values buffer
+     * @param buffer - the underlying byte buffer
+     * @param metaDataBuffer - the meta data buffer from which we compute the offset
+     * @return the counter values buffer
+     */
+    @Override
+    public UnsafeBuffer createGatewayCounterValuesBuffer(final ByteBuffer buffer) {
+        final int offset = endOfMetadata
+                + metaDataBuffer.getInt(gwCountersLblBuffersLengthOffset);
+        final int length = metaDataBuffer.getInt(gwCountersValueBuffersLengthOffset);
+        // Update offset in header section
+        metaDataBuffer.putInt(gwCountersValueBuffersReferenceOffset, offset);
+
+        return new UnsafeBuffer(buffer, offset, length);
+    }
+
+    /**
+     * Creates the counter labels buffer for the service identified by index
+     * @param buffer - the underlying byte buffer
+     * @param metaDataBuffer - the meta data buffer from which we compute the offset
+     * @param index identifier
+     * @return the counter labels buffer
+     */
+    @Override
+    public UnsafeBuffer createServiceCounterLabelsBuffer(final ByteBuffer buffer,
+                                                         int index) {
+        final int offset = endOfMetadata
+                + metaDataBuffer.getInt(gwCountersLblBuffersLengthOffset)
+                + metaDataBuffer.getInt(gwCountersValueBuffersLengthOffset) + index
+                * (SERVICE_COUNTER_VALUES_BUFFER_LENGTH + SERVICE_COUNTER_LABELS_BUFFER_LENGTH);
+        final int length = SERVICE_COUNTER_LABELS_BUFFER_LENGTH;
+
+        // Update offset in header section
+        metaDataBuffer.putInt(serviceRefSection + index * OFFSETS_PER_SERVICE * BitUtil.SIZE_OF_INT, offset);
+
+        return new UnsafeBuffer(buffer, offset, length);
+    }
+
+    /**
+     * Creates the counter values buffer for the service identifier by index
+     * @param buffer - the underlying byte buffer
+     * @param metaDataBuffer - the meta data buffer from which we compute the offset
+     * @param index - service identifier
+     * @return the counter values buffer
+     */
+    @Override
+    public UnsafeBuffer createServiceCounterValuesBuffer(final ByteBuffer buffer,
+                                                         int index) {
```

Comment:
Just to check my understanding, index is the zero-based service index, 0 for the first service, 1 for the next, etc, right?